### PR TITLE
Small improvements to storage manager UX

### PR DESCRIPTION
**Pull request by DaughterOfMars**
## Links to any relevant issues

- Related to #105 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: sdk/CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `Account::{unspent_alias_output, unspent_foundry_output, unspent_nft_output}` methods;
 - `StrongholdAdapter::inner` method;
 
+### Changed
+
+- `StorageManager` now accepts a generic storage instead of a boxed type, and the params have been reversed;
```
Comment (thibault-martinez) [inline sdk/CHANGELOG.md]: Not a big deal but technically not needed as it's not public

Comment (DaughterOfMars) [reply sdk/CHANGELOG.md]: Looks pretty public to me, but maybe I'm missing something.

Comment (thibault-martinez) [reply sdk/CHANGELOG.md]: `StorageManager::new` is `pub(crate)` ?

Comment (DaughterOfMars) [reply sdk/CHANGELOG.md]: 😆 I completely missed that

File: sdk/src/wallet/storage/manager.rs [hunk 2]
```diff
@@ -49,12 +49,12 @@ pub struct StorageManager {
...
     pub(crate) async fn new(
-        encryption_key: Option<[u8; 32]>,
-        storage: Box<dyn StorageAdapter + Send + Sync + 'static>,
+        storage: impl StorageAdapter + Send + Sync + 'static,
+        encryption_key: impl Into<Option<[u8; 32]>> + Send,
     ) -> crate::wallet::Result<Self> {
         let mut storage = Storage {
-            inner: storage,
-            encryption_key,
+            inner: Box::new(storage) as _,
```
Comment (thibault-martinez) [inline sdk/src/wallet/storage/manager.rs:56]: Interesting

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**